### ISSUE TITLE
Bug: variable hoisting ignored in rule "css/no-invalid-properties"

### ISSUE THREAD

**Issue by Otlikov** [eslint/css#199]
### Environment

ESLint version: 9.31.0
@eslint/css version: 0.10.0
Node version: 24.4.0
npm version: 11.4.2
Operating System: macOS 15.5 (24F74)

### Which language are you using?

stylesheet

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```

</details>

```css
/* eslint css/no-invalid-properties: ["error"] */

.test {
	color: var(--myColor);
}

:root {
	--myColor: blue;
}



### What did you expect to happen?

No error as the definition of variable --myColor should be hoisted.

### What actually happened?

Can't validate with unknown variable '--myColor'

### Link to Minimal Reproducible Example

none

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

_No response_

**Comment by Otlikov** [comment]
Option `allowUnknownVariables: true` actually suppressed the error message. Then, however, I don't get any warnings about missing css variables, e.g. if I remove `--myColor` altogether.
